app/src/main/java/com/andrew/tasky/agenda/presentation/screens/agenda/AgendaFragment.kt

```diff
@@ -54,12 +50,18 @@ class AgendaFragment : Fragment(R.layout.fragment_agenda) {
         collectLatestLifecycleFlow(viewModel.calendarDateItemList) { dates ->
             miniCalendarAdapter.submitList(dates)
         }
-        collectLatestLifecycleFlow(viewModel.dateSelected) { dateSelected ->
-            setupCurrentDateSelectedTextView(dateSelected)
-        }
         collectLatestLifecycleFlow(viewModel.uiAgendaItems) { items ->
             agendaAdapter.submitList(items)
         }
+        collectLatestLifecycleFlow(viewModel.currentDateAndTimeFlow) { currentDateAndTime ->
+            setupCurrentMonthTextView(currentDate = currentDateAndTime.toLocalDate())
+            collectLatestLifecycleFlow(viewModel.dateSelected) { dateSelected ->
+                setupCurrentDateSelectedTextView(
+                    dateSelected = dateSelected,
+                    currentDate = currentDateAndTime.toLocalDate()
+                )
+            }
```

Comment:
Be careful nesting collectors, using combine is safer here. So using a combine block in the VM that combines `dateSelected` and `currentDateAndTimeFlow` and exposes the result to the UI